src/components/browser/requestOperations/Swap.tsx

```diff
@@ -0,0 +1,214 @@
+import {ActiveAccount, KeyTypes} from 'actions/interfaces';
+import {showModal} from 'actions/message';
+import {RequestSwap} from 'hive-keychain-commons';
+import React, {useEffect, useState} from 'react';
+import {MessageModalType} from 'src/enums/messageModal.enums';
+import {TransactionOptions} from 'src/interfaces/multisig.interface';
+import {getCurrency} from 'utils/hive';
+import {getAccount} from 'utils/hiveUtils';
+import {RequestId} from 'utils/keychain.types';
+import {translate} from 'utils/localize';
+import {SwapTokenUtils} from 'utils/swap-token.utils';
+import {getTokenPrecision} from 'utils/tokens.utils';
+import RequestItem from './components/RequestItem';
+import RequestOperation from './components/RequestOperation';
+import {RequestComponentCommonProps} from './requestOperations.types';
+
+type Props = {
+  request: RequestSwap & RequestId;
+} & RequestComponentCommonProps;
+
+export default ({
+  request,
+  accounts,
+  closeGracefully,
+  sendResponse,
+  sendError,
+}: Props) => {
+  const {request_id, ...data} = request;
+  const {username, startToken, endToken, amount, steps, slippage} = data;
+  const [estimateValue, setEstimateValue] = useState<number>();
+  const [balance, setBalance] = useState<number>();
+  const [balanceAfterSwap, setBalanceAfterSwap] = useState<number>();
+
+  useEffect(() => {
+    const calculateEstimate = async () => {
+      try {
+        const swapConfig = await SwapTokenUtils.getConfig();
+        if (!swapConfig) return;
+
+        const lastStep = steps[steps.length - 1];
+        if (!lastStep) return;
+
+        const precision = await getTokenPrecision(lastStep.endToken);
+        const value = Number(lastStep.estimate);
+        const fee = (value * swapConfig.fee.amount) / 100;
+        const finalValue = Number(value - fee).toFixed(precision);
+        setEstimateValue(Number(finalValue));
+      } catch (error) {
+        console.error('Error calculating estimate:', error);
+      }
+    };
+
+    calculateEstimate();
+  }, [steps]);
+
+  const getBalance = async () => {
```

Comment:
This should be extracted into a RequestBalance component that does the whole logic rather than having it here with a RequestItem. This way we can reuse it for transfer operations, etc.